File: scratch/extract_svg.mjs
import { readFileSync } from 'fs';

const content = readFileSync('E:\\Dev\\Unity Text Styles\\. Type Scale.svg', 'utf-8');

// Find all unique fill colors
const fills = new Set(content.match(/fill="(#[0-9A-Fa-f]+)"/g) || []);
console.log('Colors found:', [...fills]);

// Check for image/pattern references
const images = content.match(/<image[^>]*href="([^"]*)"[^>]*/g) || [];
console.log('\nImages:', images.length);
images.slice(0, 5).forEach(img => console.log('  Image:', img.slice(0, 200)));

// Check for filter/defs
const filters = content.match(/<filter[^>]*id="([^"]*)"/g) || [];
console.log('\nFilters:', filters);

// Check for clip-paths
const clips = content.match(/<clipPath[^>]*id="([^"]*)"/g) || [];
console.log('\nClips:', clips);

// Look for embedded raster images (base64)
const base64 = content.match(/xlink:href="data:image[^"]{0,50}/g) || [];
console.log('\nBase64 images:', base64.length);
base64.slice(0, 3).forEach(b => console.log('  ', b));

// Count path elements by fill color
const paths = content.match(/<path[^>]*fill="([^"]*)"[^>]*\/>/g) || [];
const pathsByColor = {};
paths.forEach(p => {
    const m = p.match(/fill="([^"]*)"/);
    if (m) {
        pathsByColor[m[1]] = (pathsByColor[m[1]] || 0) + 1;
    }
});
console.log('\nPaths by color:', pathsByColor);

// Find rect elements
const rects = content.match(/<rect[^>]*>/g) || [];
console.log('\nRects:', rects.length);
rects.forEach(r => console.log('  ', r.slice(0, 200)));

File: scratch/extract_svg.py
import re

with open(r'E:\Dev\Unity Text Styles\. Type Scale.svg', 'r', encoding='utf-8') as f:
    content = f.read()

# Find all unique fill colors
fills = set(re.findall(r'fill="(#[0-9A-Fa-f]+)"', content))
print('Colors found:', fills)

# Find rects (background panels)
rects = re.findall(r'<rect[^/>]*/?>', content)
for r in rects[:10]:
    print('Rect:', r)

# Find text elements  
texts = re.findall(r'<text[^>]*>(.*?)</text>', content)
print('Text elements:', len(texts))
for t in texts[:20]:
    print('  Text:', t[:100])

# Find tspan elements with font info
tspans = re.findall(r'<tspan[^>]*>([^<]*)</tspan>', content)
print('\nTspan count:', len(tspans))
for t in tspans[:30]:
    if t.strip():
        print('  Tspan:', t[:100])

# Look for font-size info
font_sizes = re.findall(r'font-size[=:]["\']*(\d+)', content)
print('\nFont sizes:', set(font_sizes))

# Look for image references
images = re.findall(r'<image[^>]*href="([^"]*)"', content)
print('\nImages:', len(images))
for img in images[:5]:
    print('  Image:', img[:100])
